torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -662,14 +652,60 @@ Tensor repeat_backward(Tensor grad, IntArrayRef repeats, IntArrayRef input_shape
   for (int64_t i = 0; i < num_unsqueezed; ++i) {
     grad = grad.sum(0, false);
   }
-  for (size_t j = num_unsqueezed; j < repeats.size(); ++j) {
+
+  const auto & grad_sizes = grad.sizes();
```

Comment:
nit: there's no need to do a reference for grad.sizes(); grad.sizes() returns an IntArrayRef which is a reference to the underlying buffer already

Reply:
Actually, this is UB: grad.sizes() returns an IntArrayRef, but we're assigning it to a reference so there's nothing that is actually owning the IntArrayRef to keep it alive

```

Reply:
I think without `&`, `auto` is going to do copy construction from `IntArrayRef`. `auto &` is actually reference to the `IntArray`